src/js/angular/core/services/connectors.service.js

```diff
@@ -13,44 +13,62 @@ ConnectorsService.$inject = ['ConnectorsRestService'];
 function ConnectorsService(ConnectorsRestService) {
 
     /**
-     * Fetches all connector types from the server.
+     * Fetches all connector types from the server for the repository with id <code>repositoryId</code> and location <code>repositoryLocation</code>.
+     * If the repository ID and repository location are not provided, the values persisted in local storage will be used {@see authentication.interceptor.js}.
+     *
+     * @param {string} repositoryId - (optional) The repository id.
+     * @param {string} repositoryLocation - (optional) The repository location.
      * @return {Promise<ConnectorTypesListModel>}
      */
-    const getConnectorTypes = () => {
-        return ConnectorsRestService.getConnectors()
+    const getConnectorTypes = (repositoryId, repositoryLocation) => {
```

Comment:
I don't like it.

You are describing logic that is happening somewhere else and can be changed.
Maybe remove this from the comment and only state that if user passes id and location, they will be used explicitly